### DISCUSSION TITLE
Consider moving `@aws-sdk/*` dependencies to peers.

### DISCUSSION THREAD

**Discussion by sebpaulus** [sensedeep/dynamodb-onetable#473] [General]
The library is using `@aws-sdk/*` declared as direct dependencies. While this is not a problem when you develop a single app, we have libraries in our project that will not function properly in conjunction with `dynamodb-onetable` because yarn/npm will install multiple copies of the `aws-sdk` libs. 

Can you please consider moving the AWS SDK related items to `peerDependencies`?

**Comment by mobsense** [comment]
Can you explain a bit more how the difference between peerDependencies works in your app and how multiple copies impacts you.

The deps:

```
        "@aws-sdk/client-dynamodb": "^3.17.0",
        "@aws-sdk/util-dynamodb": "^3.17.0",
```

are used at runtime. 

Can changing to peerDependencies impact other users?

**Comment by sebpaulus** [comment]
Moving it to peers would require me to provide those libraries, but that helps with deduping the `@aws-sdk/` libs and resolving issues with AWS SDK internals (some transitive dependencies are not matching). 
In the end I was able to resolve that with peering the dynamodb-onetable as well, so we can park this discussion or close it, unless there'll be more people who will bring it up.

In the end it probably does not affect standard project and more libraries, so it could not be a big issue.

From my point of view I have found a satisfactory solution.

**Comment by mobsense** [comment]
Thanks for the feedback. Removing as a dep would require other users to manually install the SDK which is less than ideal.

Your solution seems the best.

Thank you for following up.
